lib/config/decrypt/bcpgp.ts

```diff
@@ -0,0 +1,65 @@
+import { decrypt } from '@renovatebot/pgp';
+import { logger } from '../../logger';
+import { getEnv } from '../../util/env';
+import { regEx } from '../../util/regex';
+
+export async function tryDecryptBcPgp(
+  privateKey: string,
+  encryptedStr: string,
+): Promise<string | null> {
+  try {
+    const startBlock = '-----BEGIN PGP MESSAGE-----\n\n';
+    const endBlock = '\n-----END PGP MESSAGE-----';
+    let armoredMessage = encryptedStr.trim();
+
+    const hasStartHeader = armoredMessage.startsWith(startBlock);
+    const hasEndHeader = armoredMessage.endsWith(endBlock);
+
+    if (
+      !hasStartHeader &&
+      !hasEndHeader &&
+      !armoredMessage.includes('=') &&
+      !armoredMessage.includes('\n') &&
+      armoredMessage.length % 4 !== 0
+    ) {
+      logger.debug('Adding base64 padding to armored message');
+      armoredMessage += `=`.repeat(4 - (armoredMessage.length % 4));
+    }
+
+    if (!hasStartHeader) {
+      armoredMessage = `${startBlock}${armoredMessage}`;
+    }
+    if (!hasEndHeader) {
+      armoredMessage = `${armoredMessage}${endBlock}`;
+    }
+
+    const data = await decrypt(
+      privateKey.replace(regEx(/\n[ \t]+/g), '\n'), // little massage to help a common problem
```

Comment:
No test coverage on this - would be good if we can get an example of what this actually does / fixes